internal/file/watch/watch_test.go

```diff
@@ -255,68 +245,342 @@ func Test_watch_Start(t *testing.T) {
+					t.Helper()
+					w, err := fsnotify.NewWatcher()
+					if err != nil {
+						t.Fatal(err)
+					}
+					w.Events = make(chan fsnotify.Event, 1)
+
+					w.Events <- fsnotify.Event{
+						Name: "watch.go",
+						Op:   fsnotify.Remove,
+					}
+
+					return fields{
+						w:  w,
+						eg: errgroup.Get(),
+						onDelete: func(ctx context.Context, name string) error {
+							if got, want := name, "watch.go"; got != want {
+								t.Errorf("onWrite name got = %s, want %s", got, want)
+							}
+							return errors.New("err")
+						},
+					}
+				},
+				afterFunc: func(args) {
+					cancel()
+				},
+				want: want{
+					want: func() chan error {
+						ch := make(chan error, 1)
+						ch <- errors.New("err")
+						close(ch)
+						return ch
+					}(),
+					err: nil,
+				},
+			}
+		}(),
+
+		func() test {
+			ctx, cancel := context.WithCancel(context.Background())
+			return test{
+				name: "returns (chan error, nil) but channel contains error when w.Chmod returns error",
+				args: args{
+					ctx: ctx,
+				},
+				fieldsFunc: func(t *testing.T) fields {
+					t.Helper()
+					w, err := fsnotify.NewWatcher()
+					if err != nil {
+						t.Fatal(err)
+					}
+					w.Events = make(chan fsnotify.Event, 1)
+
+					w.Events <- fsnotify.Event{
+						Name: "watch.go",
+						Op:   fsnotify.Chmod,
+					}
+
+					return fields{
+						w:  w,
+						eg: errgroup.Get(),
+						onChmod: func(ctx context.Context, name string) error {
+							if got, want := name, "watch.go"; got != want {
+								t.Errorf("onWrite name got = %s, want %s", got, want)
+							}
+							return errors.New("err")
+						},
+					}
+				},
+				afterFunc: func(args) {
+					cancel()
+				},
+				want: want{
+					want: func() chan error {
+						ch := make(chan error, 1)
+						ch <- errors.New("err")
+						close(ch)
+						return ch
+					}(),
+					err: nil,
+				},
+			}
+		}(),
+
+		func() test {
+			ctx, cancel := context.WithCancel(context.Background())
+			return test{
+				name: "returns (chan error, nil) but channel contains error when w.Rename returns error",
+				args: args{
+					ctx: ctx,
+				},
+				fieldsFunc: func(t *testing.T) fields {
+					t.Helper()
+					w, err := fsnotify.NewWatcher()
+					if err != nil {
+						t.Fatal(err)
+					}
+					w.Events = make(chan fsnotify.Event, 1)
+
+					w.Events <- fsnotify.Event{
+						Name: "watch.go",
+						Op:   fsnotify.Rename,
+					}
+
+					return fields{
+						w:  w,
+						eg: errgroup.Get(),
+						onRename: func(ctx context.Context, name string) error {
+							if got, want := name, "watch.go"; got != want {
+								t.Errorf("onWrite name got = %s, want %s", got, want)
+							}
+							return errors.New("err")
+						},
+					}
+				},
+				afterFunc: func(args) {
+					cancel()
+				},
+				want: want{
+					want: func() chan error {
+						ch := make(chan error, 1)
+						ch <- errors.New("err")
+						close(ch)
+						return ch
+					}(),
+					err: nil,
+				},
+			}
+		}(),
```

Comment:
thinking we need a test case of 
1. keep watching the file change for multiple times
2. (not sure) create a test file and actually change the file.

I am thinking this test case is not effective as it doesn't really watching the file, It just creating a fake event and test the behavior of that fake event.

Reply:
about the first case, what if the file change for multiple times with the same event?